py-polars/debug/launch.py

```diff
@@ -0,0 +1,51 @@
+import re
+import subprocess
+import sys
+import time
+from pathlib import Path
+
+
+def launch_debugging() -> None:
+    """
+    Debug Rust files via Python.
+
+    Determine the pID for the current debugging session, attach the Rust LLDB launcher,
+    and execute the originally-requested script.
+    """
+    if len(sys.argv) == 1:
+        return
+
+    cmd = (
+        f"ps -aux | grep 'debugpy/launcher/../../debugpy' "
+        f"| grep '{__file__}' | awk '{{print $2}}'"
+    )
+
+    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
+    output, _ = p.communicate()
+    p_status = p.wait()
+
+    # parse the output: the first should be the actual pid (the second is the Popened
+    # subprocess)
+    if p_status != 0 or output is None:
+        raise RuntimeError("Cannot get pID of current debugging session.")
+    output = output.decode()
+    if (out := re.match("^(\\d+)", output)) is None:
+        raise RuntimeError("Cannot get pID of current debugging session.")
+
+    pID = int(out.group(0))
+
+    # print to the console to allow the "Rust LLDB" routine to pick up on the signal
+    print(f"pID = {pID}")
+
+    # give the LLDB time to connect. We may have to play with this setting.
+    time.sleep(1)
+
+    # run the originally requested file
+    # we updated sys.argv so that when exec() is called, it's populated with
+    # the requested script name in sys.argv[0] and the remaining args after
+    sys.argv = sys.argv[1:]
+    exec(Path.open(sys.argv[0]).read())
```

Comment:
Right now this errors out for me. Something about trying to ensure that a string is closed. I don't understand why this would work for you but not for me. LLDB is attaching though.

![image](https://github.com/pola-rs/polars/assets/37670616/749463b9-2e7c-47a6-b133-8b63f864f1e8)

It actually works if I put a `Path` around `sys.argv[0]`

![image](https://github.com/pola-rs/polars/assets/37670616/4d8a7c0c-2da8-4976-8576-eff515d5a6fe)

Reply:
I'm on 3.11 which may be why. I just pushed an update that I think should resolve this for you.